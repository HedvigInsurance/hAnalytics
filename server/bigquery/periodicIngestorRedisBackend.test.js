const redisBackend = require("./periodicIngestorRedisBackend");
const { BigQueryDatetime } = require("@google-cloud/bigquery");
const Redis = require("ioredis-mock");

describe("periodicIngestorRedisBackend", () => {
  beforeEach((done) => {
    new Redis().flushall().then(() => done());
  });

  test("stores and consumes correctly", async () => {
    redisBackend.append({
      mock: "mock",
    });

    expect((await redisBackend.consume()).length).toEqual(1);
    expect((await redisBackend.consume()).length).toEqual(0);

    const numberOfItems = 5000;

    [...new Array(numberOfItems)].forEach(() => {
      redisBackend.append({
        mock: "mock",
      });
    });

    expect((await redisBackend.consume(numberOfItems / 2)).length).toEqual(
      numberOfItems / 2
    );
    expect((await redisBackend.consume(numberOfItems / 2)).length).toEqual(
      numberOfItems / 2
    );
  });

  test("reparses DateTime correctly", async () => {
    const dateTime = new BigQueryDatetime(new Date().toISOString());

    const mockItem = {
      mock: "mock",
      timestamp: dateTime,
    };

    redisBackend.append(mockItem);

    const list = await redisBackend.consume();

    expect(list).toEqual([mockItem]);
  });
});
