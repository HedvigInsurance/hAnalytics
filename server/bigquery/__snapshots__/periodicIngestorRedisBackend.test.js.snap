// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`periodicIngestorRedisBackend stores and consumes correctly 1`] = `
Array [
  Object {
    "mock": "mock",
    "property_experiments": Array [
      "123",
      "456",
    ],
  },
]
`;
